### PR TITLE
[Dataset] BitcoinOTCDataset

### PR DESCRIPTION
**Pull request by HuXiangkun**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/cluster_gcn/cluster_gcn.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def main(args):
         val_mask = val_mask.cuda()
         test_mask = test_mask.cuda()
 
-    print(torch.cuda.get_device_name(0))
+    # print(torch.cuda.get_device_name(0))
```
Comment (hetong007) [inline examples/pytorch/cluster_gcn/cluster_gcn.py:96]: Either remove it or control under `self.verbose`

File: python/dgl/data/bitcoinotc.py [hunk 2]
```diff
@@ -1,53 +1,90 @@
-from scipy import io
+"""  """
```
Comment (hetong007) [inline python/dgl/data/bitcoinotc.py:1]: what is this for?

File: python/dgl/data/utils.py [hunk 3]
```diff
@@ -216,8 +216,11 @@ def extract_archive(file, target_dir, overwrite=False):
     elif file.endswith('.gz'):
         import gzip
         import shutil
+        fname = os.path.basename(file)
+        makedirs(target_dir)
+        out_file_path = os.path.join(target_dir, fname[:-3])
```
Comment (hetong007) [inline python/dgl/data/utils.py:221]: Use standard utils like `os.path.splitext` instead of hacky `fname[:-3]`

File: python/dgl/data/utils.py [hunk 4]
```diff
@@ -216,8 +216,11 @@ def extract_archive(file, target_dir, overwrite=False):
     elif file.endswith('.gz'):
         import gzip
         import shutil
+        fname = os.path.basename(file)
```
Comment (hetong007) [inline python/dgl/data/utils.py:219]: Does the change in this function change the behavior for other datasets?

Comment (HuXiangkun) [reply python/dgl/data/utils.py:219]: Maybe I should do it in this specific dataset.

File: python/dgl/data/bitcoinotc.py [hunk 5]
```diff
@@ -56,14 +95,47 @@ def _load(self, filename):
...
     def is_temporal(self):
         return True
+
+    def _extract_gz(self, file, target_dir, overwrite=False):
+        if os.path.exists(target_dir) and not overwrite:
+            return
+        print('Extracting file to {}'.format(target_dir))
+        fname = os.path.basename(file)
+        makedirs(target_dir)
+        out_file_path = os.path.join(target_dir, fname[:-3])
```
Comment (hetong007) [inline python/dgl/data/bitcoinotc.py:132]: still, please avoid using `fname[:-3]`.

Comment (classicsong) [reply python/dgl/data/bitcoinotc.py:132]: At lease you should add some comment here for fnam[:-3]

**Pull request merged**